tests/integration/test_ec2.py

```diff
@@ -222,6 +222,52 @@ def test_describe_vpn_gateways_filter_by_vpc(self):
         ec2.delete_vpn_gateway(VpnGatewayId=gateway_id)
         ec2.delete_vpc(VpcId=vpc_id)
 
+    def test_describe_vpc_endpoints_with_filter(self):
+
+        ec2 = self.ec2_client
+        vpc = ec2.create_vpc(CidrBlock="10.0.0.0/16")
+        vpc_id = vpc["Vpc"]["VpcId"]
+
+        # test filter of Gateway endpoint services
+        vpc_endpoint_gateway_services = ec2.describe_vpc_endpoint_services(
+            Filters=[
+                {"Name": "service-type", "Values": ["Gateway"]},
+            ],
+        )
+
+        self.assertEqual(200, vpc_endpoint_gateway_services["ResponseMetadata"]["HTTPStatusCode"])
+        services = vpc_endpoint_gateway_services["ServiceNames"]
+        self.assertEqual(2, len(services))
```

Comment:
nit: Can we expect the length of this list always to be exactly `2`? Otherwise, we should probably remove this assertion, to remain forward-compatible with potential future changes.